pkg/controller/poolcoordinator/lister/lister.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2022 The OpenYurt Authors.
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lister
+
+import (
+	"time"
+
+	"k8s.io/client-go/informers"
+	informercoordv1 "k8s.io/client-go/informers/coordination/v1"
+	v1 "k8s.io/client-go/informers/core/v1"
+	client "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+const (
+	resyncInt = 5 * time.Second
+)
+
+type ACallback func(interface{})
+type UCallback func(interface{}, interface{})
+
+func SetInformerHandler(informer cache.SharedIndexInformer, afunc ACallback, ufunc UCallback, dfunc ACallback) {
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    afunc,
+		UpdateFunc: ufunc,
+		DeleteFunc: dfunc,
+	})
+}
+
+func CreateNodeInformer(client client.Interface, factory informers.SharedInformerFactory, stopper chan (struct{})) v1.NodeInformer {
+	nodeInformer := factory.Core().V1().Nodes()
+	factory.Start(stopper)
+	factory.WaitForCacheSync(stopper)
```

Comment:
we can not start informer at here, because all of informers will be started at https://github.com/openyurtio/openyurt/blob/master/cmd/yurt-controller-manager/app/controllermanager.go#L208

Reply:
and `factory.WaitForCacheSync(stopper)` should be executed in a independent goroutine, you can take YurtCSRApprover controller as a example: https://github.com/openyurtio/openyurt/blob/master/pkg/controller/certificates/csrapprover.go#L181